src/Symfony/Component/Security/Http/Authenticator/AccessToken/QueryAccessTokenExtractor.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Authenticator\AccessToken;
+
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Extracts a token from a query string parameter.
+ *
+ * @author Florent Morselli <florent.morselli@spomky-labs.com>
+ *
+ * @final
+ */
+class QueryAccessTokenExtractor implements AccessTokenExtractorInterface
+{
+    public const PARAMETER = 'access_token';
+    private string $parameter;
+
+    public function __construct(string $parameter = self::PARAMETER)
+    {
+        $this->parameter = $parameter;
+    }
+
+    public function extractToken(Request $request): ?string
+    {
+        if (!$request->query->has($this->parameter)) {
+            return null;
+        }
+        $parameter = $request->query->get($this->parameter);
```

Comment:
what happens if the parameter is an array ? We should not throw an exception on the get call.

Reply:
I don't see any reason for passing several access tokens at the same time.
The following line considers that only a string is valid, otherwise it returns `null`.

Reply:
I am not sure to understand why we could fall in a 500 error.
The `InputBag::get` is only called if the `has` method returned `true`.
Should I add a try/catch in the `support()` method to prevent any misuse?

Reply:
I tried to create a test with an `array` but I am facing an issue with the `InputBag` object.
With [InputBag.php#L65](https://github.com/symfony/symfony/blob/6.2/src/Symfony/Component/HttpFoundation/InputBag.php#L65) you can set an array, but cannot return it because of the `is_scalar` check at [InputBag.php#L36](https://github.com/symfony/symfony/blob/6.2/src/Symfony/Component/HttpFoundation/InputBag.php#L36)

Reply:
OK I added a [try-catch statement](https://github.com/symfony/symfony/pull/46428/files#diff-5121e3f50b05d1b7abafdb3fdeaa4cac43d1382956e7ca97e247bd985800a1aaR74-R80) for the `extractAccessToken` method in the authenticator.
Having the try-catch here will also prevent custom extractors for throwing an exception and forces having a consistent behaviour i.e. returning null or a string.
The method `authenticate` now [only throw exceptions](https://github.com/symfony/symfony/pull/46428/files#diff-5121e3f50b05d1b7abafdb3fdeaa4cac43d1382956e7ca97e247bd985800a1aaR74-R80) that implement `HttpExceptionInterface`

Reply:
While the interface only specifies `AuthenticationException` (I plan to challenge this at some point), in case the parameter is present but with a wrong value (e.g. empty string or array) the response should really be a 400 Bad Request, as already done in other authenticators (see e.g. JsonLogin). 
Also we should avoid catching any `Throwable`, any unexpected exception should actually result in a 500.

Reply:
> I don't see any reason for returning a 400 status code during the authentication process.

Because the security layer is reached before everything else, some basic input validation has to be done before actually entering the authentication process.

We did that for `json_login` in https://github.com/symfony/symfony/pull/22569, which is now located in https://github.com/symfony/symfony/blob/d09634946b0e9e7035e138b52d6260b2bf7b783c/src/Symfony/Component/Security/Http/Authenticator/JsonLoginAuthenticator.php#L168

> And in any case, you must ensure that supports does not throw an exception for unsupported requests

Maybe `supports` should not extract the token then

Reply:
> Actually I wouldn't say that the RFC clearly indicates to return a 401 in such case.

I assume this is the expected status code because of the presence of the `WWW-Authenticate` header parameter.
With that being said, the RFC does not enforce this status code or the presence of the header parameter.

> And in any case, you must ensure that supports does not throw an exception for unsupported requests

I agree with that. Best to me is to catch the exception and continue with the other token extractor.
To allow debugging as @derrabus suggested, I propose to inject a logger service and send debug messages if deemed necessary.